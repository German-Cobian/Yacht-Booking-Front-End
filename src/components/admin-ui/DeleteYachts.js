import axios from 'axios';
import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { Navigate } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Table from 'react-bootstrap/Table';
import Button from 'react-bootstrap/Button';
import './admin-ui.css';

const DeleteYachts = () => {
  const [yachts, setYachts] = useState([]);
  const { currentUser } = useSelector((state) => state.auth);

  if (currentUser.role !== 'admin') {
    return <Navigate to="/" replace />;
  }

  const request = axios.create({
    headers: {
      'Content-Type': 'application/json',
      Authorization: localStorage.getItem('token'),
    },
  });

  useEffect(() => {
    // Get Yachts from API
    request.get('https://wishyacht-api.herokuapp.com/v1/yachts')
      .then((response) => {
        setYachts(response.data);
      })
      .catch((error) => error);
  }, []);
  // Delete Yachts from API

  const handleDeleteYacht = (id) => (event) => {
    event.preventDefault();

    request.delete(`https://wishyacht-api.herokuapp.com/v1/yachts/${id}`)
      .then(() => {
        const included = [...yachts].filter((yacht) => yacht.attributes.id !== id);
        setYachts(included);
      })
      .catch((error) => error);
  };

  const yachtList = yachts.map((yacht) => (
    <tr key={yacht.attributes.name}>
      <td className="fs125 d-flex justify-content-between px-5">
        {yacht.attributes.name}
        <Button
          type="button"
          variant="info"
          onClick={handleDeleteYacht(yacht.attributes.id)}
        >
          Delete
        </Button>
      </td>
    </tr>
  ));

  return (
    <main>
      <div className="effect" />
      <Container className="align-items-center justify-content-center z1">
        <Table striped borderless hover responsive className="align-items-center my-5 mx-auto w-80 shadow p-3 bg-body rounded">
          <thead>
            <tr>
              <th className="fs125 my-2">Yachts</th>
            </tr>
          </thead>
          <tbody>
            {yachtList}
          </tbody>
        </Table>
      </Container>
    </main>
  );
};

export default DeleteYachts;
