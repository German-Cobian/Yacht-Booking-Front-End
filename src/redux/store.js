// import { legacy_createStore as createStore, applyMiddleware } from 'redux';
// import { composeWithDevTools } from 'redux-devtools-extension';
// import thunk from 'redux-thunk';
// import rootReducer from './reducers';
// import fetchYachts from './actions/yachtActions';

// export default createStore(
//   rootReducer,
//   composeWithDevTools(applyMiddleware(thunk)),
// );

// store.dispatch(fetchYachts());
