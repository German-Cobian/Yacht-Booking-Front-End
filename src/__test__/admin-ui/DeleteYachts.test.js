import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/storeSetUp';
import DeleteYachts from '../../components/admin-ui/DeleteYachts';

it ('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <DeleteYachts />
      </Router>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
} );