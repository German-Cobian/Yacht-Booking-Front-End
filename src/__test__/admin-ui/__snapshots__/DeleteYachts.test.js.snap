// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="main"
>
  <div
    className="deleteYachts__header"
  >
    <h1>
      Delete Yachts
    </h1>
    <p>
      These are all our yachts
    </p>
    <hr />
  </div>
  <div
    className="effect"
  />
  <div
    className="align-items-center justify-content-center z1 container"
  >
    <div
      className="table-responsive"
    >
      <table
        className="align-items-center my-5 mx-auto w-80 shadow p-3 bg-body rounded table table-striped table-borderless table-hover"
      >
        <thead>
          <tr>
            <th
              className="fs125 my-2"
            >
              Yachts
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</main>
`;
